typings/downshift.flow.js

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react'
+
+// @flow
+
+type A11StatusMessageOptions = {
+  highlightedIndex: number,
+  highlightedValue: any,
+  inputValue: string,
+  isOpen: boolean,
+  itemToString: (item: any) => string,
+  previousResultCount: number,
+  resultCount: number,
+  selectedItem: any,
+}
+
+type ChangeOptions = {
+  selectedItem: any,
+  previousItem: any,
+}
+
+type StateChangeOptions = {
+  highlightedIndex: number,
+  inputValue: string,
+  isOpen: boolean,
+  selectedItem: any,
+}
+
+type GetRootPropsOptions = {
+  refKey: string,
+}
+
+type GetItemPropsOptions = {
+  index: number,
+  item: any,
+}
+
+type ControllerStateAndHelpers = {
+  // prop getters
+  getRootProps: (options: GetRootPropsOptions) => any,
+  getButtonProps: (options: GetButtonPropsOptions) => any,
+  getLabelProps: (options: GetLabelPropsOptions) => any,
+  getInputProps: (options: GetInputPropsOptions) => any,
+  getItemProps: (options: GetItemPropsOptions) => any,
+
+  // actions
+  openMenu: () => void,
+  closeMenu: () => void,
+  toggleMenu: () => void,
+  selectItem: (item: any) => void,
+  selectItemAtIndex: (index: number) => void,
+  selectHighlightedItem: (index: number) => void,
+  setHighlightedItem: (index: number) => void,
+  clearSelection: () => void,
+
+  // state
+  highlightedIndex: number,
+  inputValue: string,
+  isOpen: boolean,
+  selectedItem: any,
+}
+
+type GetInputPropsOptions = React.InputHTMLAttributes
+
+type GetLabelPropsOptions = React.LabelHTMLAttributes
+
+type GetButtonPropsOptions = {
+  // actions
+  clearSelection: () => void,
+  closeMenu: () => void,
+  openMenu: () => void,
+  selectHighlightedItem: () => void,
+  selectItem: (item: any) => void,
+  selectItemAtIndex: (index: number) => void,
+  setHighlightedIndex: (index: number) => void,
+  toggleMenu: (state: boolean) => void,
+
+  // state
+  highlightedIndex: number,
+  inputValue: string,
+  isOpen: boolean,
+  selectedItem: any,
+}
+
+type ChildrenFunction = (options: ControllerStateAndHelpers) => React.Node<*>
+
+type downshiftProps = {
+  children: ChildrenFunction,
+  defaultHighlightedIndex?: number,
+  defaultSelectedItem?: any,
+  defaultInputValue?: string,
+  defaultIsOpen?: boolean,
+  getA11yStatusMessage?: (options: A11StatusMessageOptions) => any,
+  itemToString?: (item: any) => any,
+  onChange?: (
+    options: ChangeOptions,
+    stateAndHelpers: ControllerStateAndHelpers,
+  ) => void,
+  onStateChange?: (
+    options: StateChangeOptions,
+    stateAndHelpers: ControllerStateAndHelpers,
+  ) => void,
+  onClick?: Function,
```

Comment:
It looks like the TypeScript definitions are missing this prop. However, I also don't understand how Downshift is using the root `onClick` prop. @kentcdodds should downshift have the `onClick` prop? How is it getting used? For instance, I don't see how `root_handleClick` is being used.